client/components/TestRun/index.jsx

```diff
@@ -33,6 +39,42 @@ class TestRun extends Component {
         }
     }
 
+    handleSkipTestClick() {
+        // TODO: show the popup Isaac designed here
+        this.nextTest();
+    }
+
+    handleNextTestClick() {
+        // This button just goes to next test. You will only see this button when "reviewing" previously saved test results.
+        this.nextTest();
+    }
+
+    handleSaveResultClick() {
+        const { dispatch, cycleId, tests, run, userId } = this.props;
+        let resultsEl = this.testIframe.current.contentDocument.querySelector(
+            '#__ariaatharness__results__'
+        );
+
+        if (!resultsEl) {
+            // TODO: show popup here that says "You cannot save results until you complete and review the test"
+            return;
+        }
+
+        const result = JSON.parse(resultsEl.innerHTML);
```

Comment:
Is it ever possible that the contents can be malformed? Should this be wrapped in a try/catch? Also, can you do `resultsEl.innerText`?

Reply:
it's possible, but that would be a bug within ARIA-AT repo that would prevent us from being able to consume/run the tests. but I can guard anyway, we should know if that bug exists.